### DISCUSSION TITLE
vscode-json-language server and range formatting

### DISCUSSION THREAD

**Discussion by the42** [joaotavora/eglot#1108] [Ideas]
[vscode-json-ls](https://www.npmjs.com/package/vscode-json-languageserver) does not understand formatting of full buffer but only range formatting. A range extended to the whole buffer does the trick of course. My suggestion would be that in the case of `eglot-format` for vscode-js-ls this gets 'translated' into a lsp range format of the whole buffer.

**Comment by joaotavora** [comment]
Eglot doesn't do server-specific hacks.  You can ask the server developers to implement that, since it appears to have the technical capability to do so, just not the interface.
